### PR TITLE
Add multilevel lookup breaking change

### LINKED ISSUE #28836: [Breaking change]: Multi-level lookup behavior is being disabled
### Description

On Windows, framework-dependent .NET applications would search for frameworks in multiple install locations by default. In .NET 7, this behavior is disabled.

Runtime change: https://github.com/dotnet/runtime/pull/67022

### Version

.NET 7 Preview 4

### Previous behavior

A framework-dependent .NET application could search for frameworks in multiple install locations:

1. Subdirectories relative to:

    - `dotnet` executable when running the application through `dotnet`
    - `DOTNET_ROOT` environment variable (if set) when running the application through its executable (`apphost`)

2. Globally registered install location (if set) in `HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\<arch>\InstallLocation`.
3. Default install location of `%ProgramFiles%\dotnet` (or `%ProgramFiles(x86)%\dotnet` for 32-bit processes on 64-bit Windows).

This multi-level lookup behavior was enabled by default but could be disabled by setting the environment variable `DOTNET_MULTILEVEL_LOOKUP=0`.

### New behavior

An application targeting .NET 7 or above will only look in one location - the first location where a .NET installation is found. When running an application through `dotnet`, frameworks are only searched for in subdirectories relative to `dotnet`. When running an application through its executable (`apphost`), frameworks are only searched for in the first of the locations listed in the previous behavior section where .NET is found.

Applications targeting .NET 6 or below are unaffected.

### Type of breaking change

- [X] **Binary incompatible**: Existing binaries may encounter a breaking change in behavior, such as failure to load/execute or different run-time behavior.
- [ ] **Source incompatible**: Source code may encounter a breaking change in behavior when targeting the new runtime/component/SDK, such as compile errors or different run-time behavior.

### Reason for change

There has been a lot of feedback around issues caused by multi-level lookup:
- Confusion for users: application can pick a global/default install location despite running .NET from a private install
- Inconsistency between platforms (Windows versus non-Windows)
- Behavior breaks, often in automated systems: a new global .NET install can affect otherwise isolated builds/tests.
- Performance issues

### Recommended action

Make sure the required version of .NET is installed at the single .NET install location. Error messages on failure to launch include the expected location.

### Feature area

Deployment

### Affected APIs

_No response_

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #28836.

### CODE REVIEW

File: docs/core/compatibility/deployment/7.0/multilevel-lookup.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+## Previous behavior
+
+In previous versions, a framework-dependent .NET application searched for frameworks in multiple install locations on Windows:
+
+- When running the application through [dotnet](../../../tools/dotnet.md), subdirectories relative to the `dotnet` executable.
+- When running the application through its executable (`apphost`), the `DOTNET_ROOT` environment variable (if set).
+- The globally registered install location (if set) in **HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\<arch>\InstallLocation**.
+- The default install location of *%ProgramFiles%\dotnet* (or *%ProgramFiles(x86)%\dotnet* for 32-bit processes on 64-bit Windows).
+
+This multi-level lookup behavior was enabled by default but could be disabled by setting the environment variable `DOTNET_MULTILEVEL_LOOKUP=0`.
```
Comment (IEvangelist) [inline docs/core/compatibility/deployment/7.0/multilevel-lookup.md:19]: We should come back to this new environment variable, and make sure that we add it to the .NET environment variables article.

**Review by IEvangelist — APPROVED** [review]
Looks good to me, :shipit: when you're ready.

File: docs/core/compatibility/deployment/7.0/multilevel-lookup.md (hunk 1 above)
Comment (elinor-fung) [reply docs/core/compatibility/deployment/7.0/multilevel-lookup.md:19]: I think it is already in the list. But maybe it is worth noting that it is only for pre-.NET 7?

**Review by elinor-fung — APPROVED** [review]

File: docs/core/compatibility/deployment/7.0/multilevel-lookup.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## New behavior
+
+Applications that target .NET 7 or a later version only look for frameworks in one location, which is the first location where a .NET installation is found. When running an application through [dotnet](../../../tools/dotnet.md), frameworks are only searched for in subdirectories relative to the `dotnet` executable. When running an application through its executable (`apphost`), frameworks are only searched for in the first of the locations listed in the [previous behavior section](#previous-behavior) where .NET is found.
```
Comment (tdykstra) [inline docs/core/compatibility/deployment/7.0/multilevel-lookup.md:23]: >only searched for in the first of the locations listed in the [previous behavior section](#previous-behavior) where .NET is found.

The first location is "When running the application through [dotnet] ... subdirectories relative to the `dotnet` executable."

**Review by tdykstra — APPROVED** [review]
LGTM, but I noted one thing that looked potentially confusing

**Pull request merged**